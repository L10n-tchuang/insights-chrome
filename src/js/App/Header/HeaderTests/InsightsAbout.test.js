/* eslint-disable camelcase */
import React from 'react';
import { render, waitFor, act } from '@testing-library/react';
import ConnectedInsightsAbout, { InsightsAbout, Copyright } from '../InsightsAbout';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

describe('InsightsAbout', () => {
  beforeEach(() => {
    fetch.resetMocks();
    fetch.mockResponse(
      JSON.stringify({
        app_name: 'chrome',
        src_hash: 'f3d0275e021cfa46982fd9a4376bda7a5ee9e079',
        src_tag: '',
        src_branch: 'prod-stable',
        travis: { event_type: 'push' },
        build_branch: 'prod-stable',
        build_hash: 'b5b92cf36f81fec1c9abc302a365bff111b9f5b0',
        build_id: '7',
      })
    );
  });
  let globalNavData = require('../../../../../testdata/globalNav.json');

  const initialProps = {
    activeApp: 'someApp',
    appId: 'someID',
    dispatch: jest.fn(),
    onClose: jest.fn(),
    user: {
      username: 'someUser',
    },
    globalNav: [globalNavData],
  };
  it('should render correctly with modal closed', async () => {
    let props = {
      ...initialProps,
      isModalOpen: false,
    };
    await act(async () => {
      render(<InsightsAbout {...props} />);
    });
    await waitFor(() => expect(fetch).toHaveBeenCalled());
  });

  it('should render correctly with modal open', async () => {
    let props = {
      ...initialProps,
      isModalOpen: true,
    };
    await act(async () => {
      render(<InsightsAbout {...props} />);
    });
    await waitFor(() => expect(fetch).toHaveBeenCalled());
  });
});

describe('ConnectedInsightsAbout', () => {
  let initialState;
  let mockStore;
  let globalNavData = require('../../../../../testdata/globalNav.json');
  beforeEach(() => {
    mockStore = configureStore();
    initialState = {
      chrome: {
        user: {
          identity: {
            user: {},
          },
        },
        appId: 'test',
        globalNav: [globalNavData],
        activeApp: 'test',
      },
    };
  });

  it('should render correctly with just username', async () => {
    const store = mockStore(initialState);
    let container;
    let unmount;
    let console;
    await act(async () => {
      console = global.console;
      global.console = { error: jest.fn() };
      let wrapper = render(
        <Provider store={store}>
          <ConnectedInsightsAbout isModalOpen />
        </Provider>,
        { container: document.body }
      );
      container = wrapper.container;
      unmount = wrapper.unmount;
    });
    await waitFor(() => expect(fetch).toHaveBeenCalled());
    expect(container).toMatchSnapshot();
    unmount();
    global.console = console;
  });
});

describe('Copyright', () => {
  it('should render', () => {
    const { container } = render(<Copyright />);
    expect(container.querySelector('div')).toMatchSnapshot();
  });
});
