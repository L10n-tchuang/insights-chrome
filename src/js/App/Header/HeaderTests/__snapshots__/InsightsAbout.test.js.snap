// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedInsightsAbout should render correctly with just username 1`] = `
<body
  class="pf-c-backdrop__open"
>
  <div>
    <div
      class="pf-c-backdrop"
    >
      <div
        class="pf-l-bullseye"
      >
        <div
          aria-describedby="pf-about-modal-content-2"
          aria-labelledby="pf-about-modal-title-2"
          aria-modal="true"
          class="pf-c-about-modal-box ins-c-about-modal"
          role="dialog"
        >
          <div
            class="pf-c-about-modal-box__brand"
          >
            <img
              alt="Red Hat Logo"
              class="pf-c-about-modal-box__brand-image"
              src="test-file-stub"
            />
          </div>
          <div
            class="pf-c-about-modal-box__close"
          >
            <button
              aria-disabled="false"
              aria-label="Close Dialog"
              class="pf-c-button pf-m-plain"
              data-ouia-component-id="70"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 352 512"
                width="1em"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                />
              </svg>
            </button>
          </div>
          
          <div
            class="pf-c-about-modal-box__content"
            id="pf-about-modal-content-2"
          >
            <div
              class="pf-c-about-modal-box__body"
            >
              <div
                class="pf-c-content"
              >
                <div
                  class="pf-l-stack"
                  gutter="sm"
                >
                  <div
                    class="pf-l-stack__item"
                  >
                    Please include these details when opening a support case.
                    <button
                      aria-describedby="pf-tooltip-36"
                      aria-disabled="false"
                      aria-label="Copy details"
                      class="pf-c-button pf-m-plain"
                      data-ouia-component-id="71"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 448 512"
                        width="1em"
                      >
                        <path
                          d="M320 448v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V120c0-13.255 10.745-24 24-24h72v296c0 30.879 25.121 56 56 56h168zm0-344V0H152c-13.255 0-24 10.745-24 24v368c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24V128H344c-13.2 0-24-10.8-24-24zm120.971-31.029L375.029 7.029A24 24 0 0 0 358.059 0H352v96h96v-6.059a24 24 0 0 0-7.029-16.97z"
                        />
                      </svg>
                    </button>
                  </div>
                  <div
                    class="pf-l-stack__item"
                  >
                    <div
                      class="pf-c-content ins-c-page__about--modal"
                    >
                      <dl
                        class="ins-debug-info"
                        data-pf-content="true"
                      >
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          User Name
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        />
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Current Application
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          Landing
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Application Path
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          /
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Chrome Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Dashboard Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Inventory Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Remediations Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Vulnerability Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Compliance Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Cost Management Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Advisor Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Drift Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Migration Analytics Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Automation Hub Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Automation Analytics Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Policies Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Patch Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Automation Services Catalog Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Approval Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                        <dt
                          class=""
                          data-pf-content="true"
                        >
                          Sources Version
                          :
                        </dt>
                        <dd
                          class=""
                          data-pf-content="true"
                        >
                          f3d0275e021cfa46982fd9a4376bda7a5ee9e079.7
                        </dd>
                      </dl>
                    </div>
                  </div>
                  <div
                    class="pf-l-stack__item"
                  >
                    <div
                      class="ins-c-footer__traditional-nav ins-c-page__about--modal-footer"
                    >
                      <p
                        class="copyright"
                      >
                        Copyright © 2020 Red Hat, Inc.
                      </p>
                      <nav>
                        <ul>
                          <li>
                            <a
                              class="nav-link"
                              href="https://www.redhat.com/en/about/privacy-policy"
                              rel="noopener noreferrer"
                              target="_blank"
                            >
                              Privacy Policy
                            </a>
                          </li>
                          <li>
                            <a
                              class="nav-link"
                              href="https://access.redhat.com/help/terms/"
                              rel="noopener noreferrer"
                              target="_blank"
                            >
                              Terms of Use
                            </a>
                          </li>
                          <li>
                            <a
                              class="nav-link"
                              href="https://www.redhat.com/en/about/all-policies-guidelines"
                              rel="noopener noreferrer"
                              target="_blank"
                            >
                              All Policies and Guidelines
                            </a>
                          </li>
                        </ul>
                      </nav>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <p
              class="pf-c-about-modal-box__strapline"
            />
          </div>
          <div
            class="pf-c-about-modal-box__hero"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Copyright should render 1`] = `
<div
  class="ins-c-footer__traditional-nav ins-c-page__about--modal-footer"
>
  <p
    class="copyright"
  >
    Copyright © 2020 Red Hat, Inc.
  </p>
  <nav>
    <ul>
      <li>
        <a
          class="nav-link"
          href="https://www.redhat.com/en/about/privacy-policy"
          rel="noopener noreferrer"
          target="_blank"
        >
          Privacy Policy
        </a>
      </li>
      <li>
        <a
          class="nav-link"
          href="https://access.redhat.com/help/terms/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Terms of Use
        </a>
      </li>
      <li>
        <a
          class="nav-link"
          href="https://www.redhat.com/en/about/all-policies-guidelines"
          rel="noopener noreferrer"
          target="_blank"
        >
          All Policies and Guidelines
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
